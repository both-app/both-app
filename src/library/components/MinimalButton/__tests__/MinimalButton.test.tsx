import React from 'react'
import { render, fireEvent, waitForElement } from 'react-native-testing-library'
import * as Haptics from 'expo-haptics'

import { MinimalButton } from '../'

describe('MinimalButton', () => {
  const handleOnActionMock = jest.fn()

  afterEach(() => {
    jest.resetAllMocks()
  })

  test('should display the button', () => {
    const { toJSON } = render(
      <MinimalButton iconName="arrow_right" onAction={handleOnActionMock} />
    )

    expect(toJSON()).toMatchSnapshot()
  })

  test('should have an haptics return', async () => {
    const impactAsyncMock = jest.spyOn(Haptics, 'impactAsync')

    const { getByTestId } = render(
      <MinimalButton iconName="arrow_right" onAction={handleOnActionMock} />
    )

    fireEvent.press(getByTestId('minimalButton'))

    await waitForElement(() => expect(handleOnActionMock).toHaveBeenCalled())

    expect(impactAsyncMock).toHaveBeenCalled()
  })
})
