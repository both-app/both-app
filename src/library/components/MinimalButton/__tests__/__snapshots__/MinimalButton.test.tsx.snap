// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinimalButton should display the button 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="minimalButton"
>
  <SvgMock
    style={
      Object {
        "color": undefined,
      }
    }
  />
</View>
`;
